src/main/scala/it/pagopa/interop/authorizationprocess/api/impl/ClientApiServiceImpl.scala

```diff
@@ -121,42 +110,32 @@ final case class ClientApiServiceImpl(
     onComplete(result) {
       case Success(client)        => createApiClient201(client)
       case Failure(MissingUserId) =>
-        logger.error(
-          "Error in creating API client {} for consumer {}",
-          clientSeed.name,
-          clientSeed.consumerId,
-          MissingUserId
-        )
+        logger.error("Error in creating API client {}", clientSeed.name, MissingUserId)
         createApiClient401(problemOf(StatusCodes.Unauthorized, MissingUserId))
       case Failure(MissingBearer) =>
-        logger.error(
-          "Error in creating API client {} for consumer {}",
-          clientSeed.name,
-          clientSeed.consumerId,
-          MissingBearer
-        )
+        logger.error("Error in creating API client {}", clientSeed.name, MissingBearer)
         createApiClient401(problemOf(StatusCodes.Unauthorized, MissingBearer))
       case Failure(ex)            =>
-        logger.error(s"Error in creating client ${clientSeed.name} for consumer ${clientSeed.consumerId}", ex)
+        logger.error(s"Error in creating client ${clientSeed.name}", ex)
         internalServerError(problemOf(StatusCodes.InternalServerError, ClientCreationError))
     }
   }
 
-  /** Code: 200, Message: Client retrieved, DataType: Client
-    * Code: 401, Message: Unauthorized, DataType: Problem
-    * Code: 404, Message: Client not found, DataType: Problem
-    * Code: 500, Message: Internal server error, DataType: Problem
-    */
   override def getClient(clientId: String)(implicit
     contexts: Seq[(String, String)],
     toEntityMarshallerProblem: ToEntityMarshaller[Problem],
     toEntityMarshallerClient: ToEntityMarshaller[Client]
   ): Route = authorize(ADMIN_ROLE, SECURITY_ROLE, M2M_ROLE) {
     logger.info("Getting client {}", clientId)
     val result = for {
-      clientUuid <- clientId.toFutureUUID
-      client     <- authorizationManagementService.getClient(clientUuid)(contexts)
-      apiClient  <- getClient(client)
+      clientUuid     <- clientId.toFutureUUID
+      organizationId <- getClaimFuture(contexts, ORGANIZATION_ID_CLAIM).flatMap(_.toFutureUUID)
+      client         <- authorizationManagementService
+        .getClient(clientUuid)(contexts)
+        .ensureOr(client => InvalidOrganization(s"client $clientId", client.consumerId.toString()))(
+          _.consumerId == organizationId
+        )
```

Comment:
not sure this should be enforced.
Potentially, can a producer request information about the consumer client? @beetlecrunch

Reply:
We might ask @ruggerocastagnola if there's a UI that renders the client towards your services (you, the producer).

Reply:
Actually I was thinking more about M2M side

Reply:
Let's wait for @beetlecrunch so. Reverting is as fast as deleting 3 lines, after all.

Reply:
This means that I have to contact the party to gather the relationships. Thank you.

Reply:
@beetlecrunch